lib/src/stream_provider.dart

```diff
@@ -27,9 +49,24 @@ class StreamProvider<T>
     this.child,
   }) : super.value(key: key, value: stream);
 
+  /// {@macro provider.streamprovider.initialdata}
   final T initialData;
+
+  /// The widget that is below the current [StreamProvider] widget in the
+  /// tree.
+  /// {@macro flutter.widgets.child}
   final Widget child;
+
+  /// Optional function used whenever the listened [Stream] emits an error.
+  ///
+  /// [orElse] will be called with the emitted error and
+  /// is expected to return a fallback value without throwing.
+  ///
+  /// The returned value will then be exposed to the descendants of [StreamProvider]
+  /// like any valid value.
```

Comment:
Just wondering if this should be called `onError` or `catchError` or something with `error` in the name? `orElse` sounds a bit weird here.

Reply:
I like `catchError`. 👌